src/utils.js

```diff
@@ -17,6 +18,15 @@ module.exports = {
 // Utils
 /**************************/
 
+function firstDefined() {
+	for(var i = 0; i < arguments.length; i++) {
+		if(typeof arguments[i] !== 'undefined') {
+			return arguments[i];
+		}
+	}
+	return (function(x) { return x; })();
```

Comment:
yeah this line is actually not needed. when a function returns without any variable, the return value would be undefined.

Also minor code styles in this pr, I typically have a space after if eg. `if (bla)`. LGTM apart from that.